engine/src/retrier.rs

```diff
@@ -405,7 +405,13 @@ where
 							// We avoid small delays by always having a time of at least half.
 							let half_max = max_sleep_duration(initial_request_timeout, attempt) / 2;
 							let sleep_duration = half_max + rand::thread_rng().gen_range(Duration::default()..half_max);
-							tracing::error!("Retrier {name}: Error for request `{request_log}` with id `{request_id}`, attempt `{attempt}`: {e}. Delaying for {}ms", sleep_duration.as_millis());
+
+							let error_message = format!("Retrier {name}: Error for request `{request_log}` with id `{request_id}`, attempt `{attempt}`: {e}. Delaying for {}ms", sleep_duration.as_millis());
```

Comment:
Nit: `Duration`'s implementation of `Debug` is actually quite good, printing in human readable format, converting to seconds for long duration etc (e.g. see [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f90cc8163855cbc791c87aa5fe83ef4e)). Should we prefer that to converting to milliseconds manually?

Reply:
I think it makes it a bit easier to search the messages, so it doesn't change `ms` to `seconds` then to `minutes` etc. it'll just always be millis.

Reply:
Discussed with Alastair, and he said there were some real issues when trying to read the logs when debugging one of the issues. And upon reflection, I think ultimately while log searching is a valid concern, in this case, it'd be rare to be searching for particular "ms" values in the logs.